src/components/structures/RoomView.js

```diff
@@ -143,17 +150,46 @@ module.exports = React.createClass({
             }
         });
 
+        if (this.props.roomAddress[0] == '#') {
+            // we always look up the alias from the directory server:
```

Comment:
Discussed IRL: `MatrixClientPeg.get().getRoomIdForAlias` works for remote aliases